File: axiom-annotations/axiom_annotations_py/axiom_annotations/exceptions.py
"""Exceptions raised by axiom-annotations markers and decorators."""

from typing import Optional


class StepValidationError(Exception):
    """Raised when a Step() or AutoStep() marker has invalid arguments.

    This fires at runtime (when the function is called), giving immediate
    feedback in notebooks and during execution.
    """

    def __init__(self, step_num: Optional[float] = None, message: str = ""):
        self.step_num = step_num
        prefix = f"Step {step_num}" if step_num is not None else "Step"
        super().__init__(f"{prefix}: {message}")

File: axiom-annotations/axiom_annotations_js/src/exceptions.ts
/**
 * Exceptions raised by axiom-annotations markers and decorators.
 */

/**
 * Raised when a Step() or AutoStep() marker has invalid arguments.
 *
 * Fires at runtime (when the marker call is evaluated), giving immediate
 * feedback during execution and in REPLs / notebooks.
 */
export class StepValidationError extends Error {
  public readonly stepNum: number | undefined;

  constructor(stepNum: number | undefined, message: string) {
    const prefix = stepNum !== undefined ? `Step ${stepNum}` : "Step";
    super(`${prefix}: ${message}`);
    this.name = "StepValidationError";
    this.stepNum = stepNum;
    Object.setPrototypeOf(this, StepValidationError.prototype);
  }
}
